spec/unit/util_spec.rb

```diff
@@ -871,6 +871,23 @@ def withenv_utf8(&block)
       end
     end
 
+    it "should use a temporary staging location if provided" do
+      new_target = target.path + '.baz'
+      expect(Puppet::FileSystem.exist?(new_target)).to be_falsey
+
+      temp_target = Puppet::FileSystem::Uniquefile.new(Puppet::FileSystem.basename_string(new_target), Dir.tmpdir())
+
+      begin
+        expect(Puppet::FileSystem::Uniquefile).to receive(:new)
+          .with(Puppet::FileSystem.basename_string(new_target), Dir.tmpdir())
+          .and_return(temp_target)
+        subject.replace_file(new_target, 0555, Dir.tmpdir()) {|fh| fh.puts "foo" }
+      ensure
+        Puppet::FileSystem.unlink(new_target) if Puppet::FileSystem.exist?(new_target)
+        Puppet::FileSystem.unlink(temp_target) if Puppet::FileSystem.exist?(temp_target.to_s)
```

Comment:
If I'm correctly understanding what this test is trying to do, I think we can simplify it to something like the following?

```ruby
    it "should use a temporary staging location if provided" do
      new_target = File.join(tmpdir('new_file'), 'new_file.baz')
      staging_target = tmpdir('staging_file')

      subject.replace_file(new_target, 0555, staging_target) do |fh|
        expect(File.dirname(fh.path)).to eq(staging_target)

        fh.puts "foo"
      end
    end
```

Reply:
The `ensure` block that's removing the temp files shouldn't be necessary any more. Part of what `tmpdir` does is add the directories it creates to a list that gets cleaned up at the end of each test.